envpool/classic_control/cartpole.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2021 Garena Online Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ENVPOOL_CLASSIC_CONTROL_CARTPOLE_H_
+#define ENVPOOL_CLASSIC_CONTROL_CARTPOLE_H_
+
+#include <cmath>
+#include <random>
+
+#include "envpool/core/async_envpool.h"
+#include "envpool/core/env.h"
+
+namespace classic_control {
+
+class CartPoleEnvFns {
+ public:
+  static decltype(auto) DefaultConfig() {
+    return MakeDict("max_episode_steps"_.bind(200),
+                    "reward_threshold"_.bind(195.0));
+  }
+  template <typename Config>
+  static decltype(auto) StateSpec(const Config& conf) {
+    // TODO(jiayi): specify range with [4.8, fmax, np.pi / 7.5, fmax]
```

Comment:
@mavenlin

Reply:
https://github.com/sail-sg/envpool/blob/master/envpool/core/spec.h#L64
Need to change a more general data structure for taking in a nested list.

Reply:
I think we can open another PR to support this feature instead of this one, right?

Reply:
Yep, marking this as resolved. Opened an issue at https://github.com/sail-sg/envpool/issues/27